Added Dynamic Row Default Example (#7918)
magento/devdocs: src/guides/v2.3/ext-best-practices/tutorials/dynamic-row-system-config.md

This block sets values for the drop-down option.

> ## Step 4: Set default values - **OPTIONAL**

Author: Fixed

It is possible to set defaults for a dynamic row configuration, this is done by adding additional XML to the defaults block in the `config.xml` file for the module.

> Add a block to the `<default>` section in the file as it would be done normally but do not add any values. For the example this tutorial has been using, this would look like :

Reviewer: I recommend to cut the fluff. Less chance of deviating from the tutorial at hand.
Also, it doesn't hurt to reiterate the filename in case they skimmed over the previous paragraph and didn't quite catch that.

```suggestion
Add a block to the `<default>` section of the `config.xml` file and do not include any values:
```

Author: Fluff - and I thought I was being nice and clear 😄

```xml
<system>
    <general>
        <ranges></ranges>
    </general>
</system>
```

For each complete line of configuration, create an XML block with a repeating node that has a different value to all the others and contains XML tags for each sub-option and value.

> So we could use `<item1>, <item2>` etc.

Reviewer: ```suggestion
For example, you can use `<item1>, <item2>`.
```

The sub-options are the columns defined in the `_prepareToRender()` method as described in [Step 2](#step-2-create-the-block-class-to-describe-custom-field-columns).

> So for a single row for above we may have :

Reviewer: ```suggestion
In the following excerpt, a single row for `item1` contains 4 sub-options:
```

```xml
<item1>
    <from_qty>5</from_qty>
    <to_qty>6</to_qty>
    <price>10.00</price>
    <tax>1</tax>
</item1>
```

> Putting this all together the default block for the configuration option 'ranges', with 3 values could look like :

Reviewer 2: I would suggest mentioning one more time that this example represents the `config.xml` file. I'm pretty sure that there will be developers who try to add it to the `system.xml` 🙂

Reviewer: I agree, as I suggested in an earlier comment. 👍 

For this sentence, I think it best to be very specific.

```suggestion
Continue building the default block by adding 3 items to the `ranges` configuration option:
```

Author: Will combine both suggestions

> 

Reviewer: For this code block, you can use the convention (used earlier in this topic) of indicating in which file to add the code. I think using `>`

```xml
<system>
    <general>
        <ranges>
            <item1>
                <from_qty>1</from_qty>
                <to_qty>5</to_qty>
                <price>10.00</price>
                <tax>1</tax>
            </item1>
            <item2>
                <from_qty>6</from_qty>
                <to_qty>10</to_qty>
                <price>20.00</price>
                <tax>1</tax>
            </item2>
            <item3>
                <from_qty>11</from_qty>
                <to_qty>15</to_qty>
                <price>30.00</price>
                <tax>0</tax>
            </item3>
        </ranges>
    </general>
</system>
```

> To test, you just need to ensure that this configuration option has no entry in the database and clear the `config` cache. When the Admin section displaying your configuration is refreshed, the default values will appear.

Author: Going to turn this into a list for clarity

In the following excerpt, a single row for `item1` contains 4 sub-options:

```xml
<item1>
    <from_qty>5</from_qty>
    <to_qty>6</to_qty>
    <price>10.00</price>
    <tax>1</tax>
</item1>
```

Continue building the default block by adding 3 items to the `ranges` configuration option in the `config.xml` file:

```xml
<system>
    <general>
        <ranges>
            <item1>
                <from_qty>1</from_qty>
                <to_qty>5</to_qty>
                <price>10.00</price>
                <tax>1</tax>
            </item1>
            <item2>
                <from_qty>6</from_qty>
                <to_qty>10</to_qty>
                <price>20.00</price>
                <tax>1</tax>
            </item2>
            <item3>
                <from_qty>11</from_qty>
                <to_qty>15</to_qty>
                <price>30.00</price>
                <tax>0</tax>
            </item3>
        </ranges>
    </general>
</system>
```

To verify the default values for the configuration are correct, do the following :

> -  Ensure that this configuration option has no entry in the database.

Reviewer: Does this require any explanation as to how to verify? Just checking...

## Result

> The result is a new dynamic system row field in the Admin panel. If you have set optional default values these should also appear.

Reviewer: ```suggestion
The result is a new dynamic system row field in the Admin panel. If you have set optional default values, these should also appear.
```
